src/hotspot/share/runtime/threads.cpp

```diff
@@ -526,6 +526,9 @@ jint Threads::create_vm(JavaVMInitArgs* args, bool* canTryAgain) {
   // Initialize global data structures and create system classes in heap
   vm_init_globals();
 
+  // Once mutexes are ready, we can use NMT locks.
+  MemTracker::set_done_bootstrap();
```

Comment:
This should be done after the main thread has attached and set the current thread, otherwise if we hit any NMT code that needs the lock it would crash trying to acquire it.

Reply:
Ok I've moved it until after `initialize_thread_current`.